src/voting-tools-rs/src/logic/mod.rs

```diff
@@ -123,7 +123,7 @@ fn convert_to_snapshot_entry(
         .get(&stake_key)
         .ok_or_else(|| eyre!("no voting power available for stake key: {}", stake_key))?;
 
-    let voting_power = voting_power.clone();
+    let voting_power = voting_power.to_u128().unwrap_or(0);
```

Comment:
Do we want to panic here? Admittedly overflowing a u128 is unlikely, but it probably indicates a bug, and we should probably panic in that case

Reply:
No, I don't want ANY avoidable panics in this tool.
Because the consumer is an automated process and it can not deal with it.
Things will just catastrophically break.
IF we calculated voting power down to Lovelace,  Maximum supply is 45,000,000,000 ADA * 1,000,000 Lovelace.
Which is way smaller than a U64.
I don't like bigdecimal because it feels like deciding not to understand the limits of the system your building, and all systems have limits.  its also a "real" number and we don't have fractional components in voting power.  We should limit ourselves to integers, and if someone wants voting power down to "Lovelace" then that the base unit.

So U64 would be totally safe.
But U128 is trivial and fast to deal with on any modern computer, even SBC's like the raspberry PI.
So, I would be totally comfortable with all processing of these numbers being u128.
No one is making a currency with U128 worth of voting power.
Checking ranges and validity is outside the scope of this tool.
If we use U128, then this conversion is unnecessary.
The only way this could fail to convert (I guess) is if its some really small value < 0.  Not sure what bigdecimal does on that conversion to u128.  But in any event, 0 should be the result.
I would not change it for this tool,  not would I change BigDecimal because it works.  But we should not use it anywhere else without very very good reason.